### PR TITLE
chore: disable legacy tests on branch

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
We decided we don't want to run `e2e-legacy` on the branch.

### Checklist prior to review

- [ ] Link to docs PR or issue --
- [ ] Link to UI issue or PR --
- [ ] Is the [issue worked on linked][1]? --
- [ ] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [ ] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -288,25 +288,38 @@ jobs:
...
+            if [[ $(./tools/ci/has_label.sh ci/run-full-matrix) == "true" ]]; then
+              exit 0
+            fi
+            
+            # Handle legacy tests on branch
+            if [[ "<< parameters.target >>" == "test/e2e-legacy" && "<< pipeline.git.branch >>" != "master" && "<< pipeline.git.branch >>" != "release-"* ]]; then
+              if [[ ! $(./tools/ci/has_label.sh ci/run-e2e-legacy) == "true" ]];
+                skip "we do not run legacy E2E on branch by default. To run them add ci/run-e2e-legacy label"
+              fi
+              if [[ "<< parameters.cniNetworkPlugin >>" == "calico" || "<< parameters.k8sVersion >>" == "kindIpv6" || "<< parameters.k8sVersion >>" == "v1.19.16-k3s1" || "<< parameters.arch >>" == "arm64" ]]; then
```
Comment (lahabana) [inline .circleci/config.yml:318]: Should we just remove this now? (when adding the label we should probably run the other too)

Comment (jakubdyszkiewicz) [reply .circleci/config.yml:318]: It's either
* `run-e2e-legacy` to just run one variant of E2E (the one we were running so far)
* `run-full-matrix` to run everything 

Do you think `run-e2e-legacy` is not useful and we just use `run-full-matrix` ?

Comment (lahabana) [reply .circleci/config.yml:318]: Yes that was my thinking (would people really bother making a difference?)

Comment (jakubdyszkiewicz) [reply .circleci/config.yml:318]: Ok, I gather more opinions. I'm dropping the `run-e2e-legacy`

**Review by Automaat — APPROVED** [review]

**Pull request merged**